Source/VCGeneration/Splits/BlockRewriter.cs

```diff
@@ -35,67 +38,112 @@ public Cmd TransformAssertCmd(Cmd cmd) {
     return cmd;
   }
 
-  public IEnumerable<ManualSplit> GetSplitsForIsolatedPaths(Block lastBlock, IReadOnlySet<Block>? blocksToInclude,
-    IToken origin) {
-    var blockToVisit = new Stack<ImmutableStack<Block>>();
-    var newToOldBlocks = new Dictionary<Block, Block>();
-    var newLastBlock = Block.ShallowClone(lastBlock);
-    newLastBlock.Predecessors = lastBlock.Predecessors;
-    blockToVisit.Push(ImmutableStack.Create(newLastBlock));
-    newToOldBlocks[newLastBlock] = lastBlock;
-
-    while (blockToVisit.Any()) {
-      var path = blockToVisit.Pop();
-      var firstBlock = path.Peek();
-      IEnumerable<Block> predecessors = firstBlock.Predecessors;
-      if (blocksToInclude != null) {
-        predecessors = predecessors.Where(blocksToInclude.Contains);
-      }
 
-      var hadPredecessors = false;
-      foreach (var oldPrevious in predecessors) {
-        hadPredecessors = true;
-        var newPrevious = Block.ShallowClone(oldPrevious);
-        newPrevious.Predecessors = oldPrevious.Predecessors;
-        newPrevious.Cmds = oldPrevious.Cmds.Select(TransformAssertCmd).ToList();
-
-        newToOldBlocks[newPrevious] = oldPrevious;
-        if (newPrevious.TransferCmd is GotoCmd gotoCmd) {
-          newPrevious.TransferCmd =
-            new GotoCmd(gotoCmd.tok, new List<string> { firstBlock.Label }, new List<Block> {
-              firstBlock
-            });
-        }
+  /// <summary>
+  /// Each focus block creates two options.
```

Comment:
nit: focus is probably the wrong word here. Or else there should be a more explicit statement that isolating paths is implemented in terms of `{:focus}`

Is it correct to think of `{:isolate "paths"}` as equivalent to adding `{:focus}` to each branch?

Also how does path-based splitting work on while loops?

Reply:
> focus is probably the wrong word here

Comment was accidentally copied

> Is it correct to think of {:isolate "paths"} as equivalent to adding {:focus} to each branch?

I have not managed to think of one entirely in terms of the other. I wanted to see if I could reuse the focus code for either `{:isolate "paths}` on asserts or on returns, but didn't manage.

> Also how does path-based splitting work on while loops?

A loop creates a branch that does not converge, so path-based splitting is not useful to process it.

```
LoopHead:
 assert <invariants>
 goto LoopDone, LoopBody

LoopBody:
  havoc loopTargets;
  assume <invariants>;
  assume <guard>;
  <body>
  assert <invariants>; // adding {:isolate "paths"} here does not help, since only a single path leads here
  return;

LoopDone:
  assume !<guard>
  <rest>
  assert <foo> // adding {:isolate "paths"} here does not help, since only a single path leads here
```